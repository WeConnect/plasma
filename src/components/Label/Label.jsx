// @flow
import React, { type Node } from 'react';
import cx from 'classnames';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

const types = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  DISABLED: 'disabled',
};

type Props = {|
  text: string,
  type: string,
  children?: Node,
  data?: Data,
  className?: string,
|};

const Label = ({ type, text, data, children }: Props) => {
  const labelStyle = cx(style.label, {
    [style.labelSecondary]: type === types.SECONDARY,
    [style.labelDisabled]: type === types.DISABLED,
  });
  return (
    <div {...getDataAttrs(data)} className={labelStyle}>
      {text || children}
    </div>
  );
};

Label.defaultProps = {
  type: types.PRIMARY,
};

Label.displayName = 'Plasma@Label';

export default Label;
