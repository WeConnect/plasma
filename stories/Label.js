import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import Label from '../src/components/Label/Label';

storiesOf('Label', module).add('primary', () => <Label text="A label" />);

storiesOf('Label', module).add('secondary', () => (
  <Label type="secondary" text="A secondary label" />
));

storiesOf('Label', module).add('disabled', () => <Label type="disabled" text="A disabled label" />);
