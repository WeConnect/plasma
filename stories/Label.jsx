import React from 'react';
import { storiesOf } from '@storybook/react';
import Label from '../src/components/Label/Label';
import TextInput from '../src/components/TextInput/TextInput';

const stories = storiesOf('Label', module);

stories.add('default', () => <Label>Label</Label>);

stories.add('with input', () => (
  <div>
    <Label htmlFor="text">Label</Label>
    <TextInput id="text" />
  </div>
));

stories.add('required', () => (
  <div>
    <Label required htmlFor="text">
      Label
    </Label>
    <TextInput id="text" />
  </div>
));

stories.add('disabled', () => (
  <div>
    <Label disabled htmlFor="text">
      Label
    </Label>
    <TextInput id="text" disabled />
  </div>
));

stories.add('inline', () => (
  <div>
    <Label htmlFor="text" inline>
      Label
    </Label>
    <TextInput id="text" style={{ display: 'inline-block', width: '30em' }} />
  </div>
));
