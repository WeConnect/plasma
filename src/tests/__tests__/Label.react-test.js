/* eslint-disable */

import React from 'react';
import Label from '../../components/Label/Label';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Label Component', () => {
  test('render', () => {
    const component = renderer.create(<Label />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
