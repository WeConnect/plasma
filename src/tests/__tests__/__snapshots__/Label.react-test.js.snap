// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component render 1`] = `
<div
  className="label"
/>
`;
